Refactor ReplaceItemCommand class

theelicht opened the pull request:
Hii!

It's been quite a while since I was last active on this project. I decided to check up on your progress after all this time and noticed the lack of code quality in what I wrote. I decided to refactor it to be easier to read. I verified the changes by doing a text compare before and after the change.

Greetings,
Theelicht

bradytheinventor commented:
Looks good to me, especially because this class is relatively simple. In the spirit of refactoring, however, I will point out your `_generate_command` encodes a lot of functionality and might still be hard to understand. You might consider writing several "generate" functions and then using `new_format` to select which one is called. That would break the logic apart (old command format in its own function, new command format in its own function) while still being cleaner than the if-else tree you had before.

theelicht commented:
I agree with you on having the old & new format split in different functions. I changed it and also refactored the main method to remove duplicate code.
